Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -2557,6 +2560,24 @@ def visit_CClassDefNode(self, node):
         self.in_py_class = old_in_pyclass
         return node
 
+    def _body_without_directives(self, node, directives_to_remove):
+        # For things like cclass, cfunc, etc the directives
+        # should not keep propagating to nested funcs/classes
+        # after they have been applied. This function creates
+        # a CompilerDirectivesNode to ensure that
+        new_directives = Options.copy_inherited_directives(self.directives)
+        for d in directives_to_remove:
+            del new_directives[d]
```

Comment:
IMHO, they shouldn't be deleted but reset.

Did you consider letting `copy_inherited_directives` do this? Maybe even in general?

Reply:
This particular directives look to be distinguished by whether they exist or not (`if 'cclass' in self.directives`) so I think deletion is the right thing to do.

The trouble with `copy_inherited_directives` is that it completely breaks the "with statement" form (`with cython.cfunc`) because they're automatically removed from any function in the with statement.